### PR TITLE
Feature/allow complete custom string cx 1747

### PR DESCRIPTION
**Pull request by stefaniacardenas**
# Problem
The `complete` step cannot be customised by passing strings within the step options. Discussion [here](https://github.com/onfido/onfido-sdk-ui/issues/297#issuecomment-359375383)

# Solution
Added support for custom strings passed within the step options

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [n/a] Have new automated tests been implemented?
- [n/a] Have new manual tests been written down?
- [x] Have tests passed locally?
- [n/a] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-315-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-315-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Router/StepComponentMap.js [hunk 1]
```diff
@@ -39,9 +40,10 @@ const createDocumentComponents = (documentType) => {
   return frontDocumentFlow
 }
 
-const crossDeviceSteps = (hasComplete) => {
+const crossDeviceSteps = (steps) => {
   const baseSteps = [{'type': 'crossDevice'}]
-  return hasComplete ? [...baseSteps, {'type': 'complete'}] : baseSteps
+  const completeStep = steps.find((step) => step.type === 'complete')
+  return hasCompleteStep(steps) ? [...baseSteps, completeStep] : [...baseSteps]
```
Comment (stephencookdev) [inline src/components/Router/StepComponentMap.js]: why `[...baseSteps]`? Not really a big deal, but it's inconsistent with the other methods - and looks unnecessary? (i.e. just `baseSteps` instead would be sufficient, no?)

File: src/components/Complete/index.js [hunk 2]
```diff
@@ -10,11 +10,13 @@ class Complete extends Component {
     this.props.nextStep()
   }
 
-  render ({i18n}) {
+  render ({message, submessage, i18n}) {
+    const completeMessage = message ? message : i18n.t('complete.message')
+    const completeSubmessage = submessage ? submessage : i18n.t('complete.submessage')
```
Comment (stephencookdev) [inline src/components/Complete/index.js]: pretty minor, but these could just be `x = x || i18n.t('...')` instead, right? Save a bit of typing

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-315-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-315-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://315-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Router/StepComponentMap.js [hunk 3]
```diff
@@ -10,15 +10,16 @@ import CrossDeviceLink from '../crossDevice/CrossDeviceLink'
...
+  const captureSteps = mobileFlow ? clientCaptureSteps(steps) : steps
   return flow === 'captureSteps' ?
     createComponentList(captureStepsComponents(documentType, mobileFlow), captureSteps) :
-    createComponentList(crossDeviceComponents, crossDeviceSteps(hasComplete))
+    createComponentList(crossDeviceComponents, crossDeviceSteps(steps))
 }
 
-const clientCaptureSteps = (steps, hasComplete) =>
-  hasComplete ? steps : [...steps, {type: 'complete'}]
+const hasCompleteStep = (steps) => steps.some(step => step.type === 'complete')
```
Comment (rfreitas) [inline src/components/Router/StepComponentMap.js]: Minor: `step => step.type === 'complete'` can be made into a constant `isComplete` and reused

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-315-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-315-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-315-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-315-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-315-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-315-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://315-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**